sc/memiavl/db_test.go

```diff
@@ -137,6 +137,64 @@ func TestRewriteSnapshotBackground(t *testing.T) {
 	stopped = true
 }
 
+// Ensures snapshot rewrite is triggered when current height minus last snapshot height
+// exceeds the configured snapshot interval (strictly greater).
+func TestSnapshotTriggerOnIntervalDiff(t *testing.T) {
+	dir := t.TempDir()
+	db, err := OpenDB(logger.NewNopLogger(), 0, Options{
+		Dir:                dir,
+		CreateIfMissing:    true,
+		InitialStores:      []string{"test"},
+		SnapshotInterval:   3,
+		SnapshotKeepRecent: 0,
+	})
+	require.NoError(t, err)
+
+	// helper to commit one change to bump height
+	commitOnce := func(key, val string) int64 {
+		cs := []*proto.NamedChangeSet{
+			{Name: "test", Changeset: iavl.ChangeSet{Pairs: []*iavl.KVPair{{Key: []byte(key), Value: []byte(val)}}}},
```

Comment:
Minor refactor to make this line slightly more readable.
```suggestion
		pairs := []*iavl.KVPair{{Key: []byte(key), Value: []byte(val)}}
		cs := []*proto.NamedChangeSet{
			{Name: "test", Changeset: iavl.ChangeSet{Pairs: pairs}},
		}
```